airflow/providers/smtp/notifications/smtp.py

```diff
@@ -25,6 +25,24 @@
 from airflow.notifications.basenotifier import BaseNotifier
 from airflow.providers.smtp.hooks.smtp import SmtpHook
 
+try:
+    from airflow.settings import SMTP_DEFAULT_TEMPLATED_HTML_CONTENT_PATH, SMTP_DEFAULT_TEMPLATED_SUBJECT
+except ImportError:
+    # This is a fallback for when the settings are not available - they were only added in 2.8.1,
+    # so we should be able to remove it when min airflow version for the SMTP provider is 2.9.0
+    # we do not raise deprecation warning here, because the user might be using 2.8.0 and the new provider
+    # deliberately, and we do not want to upgrade to newer version of Airflow so we should not raise the
+    # deprecation warning here. If the user will modify the settings in local_settings even for earlier
+    # versions of Airflow, they will be properly used as they will be imported above
+    SMTP_DEFAULT_TEMPLATED_HTML_CONTENT_PATH = (Path(__file__).parent / "templates" / "email.html").as_posix()
+    SMTP_DEFAULT_TEMPLATED_SUBJECT = """
+{% if ti is defined %}
+DAG {{ ti.dag_id }} - Task {{ ti.task_id }} - Run ID {{ ti.run_id }} in State {{ ti.state }}
+{% elif slas is defined %}
+SLA Missed for DAG {{ dag.dag_id }} - Task {{ slas[0].task_id }}
+{% endif %}
+"""
+
```

Comment:
Do we need this check? And could we just directly define it into the provider package without access to the airflow.settings?

E.g. `airflow.settings.SMTP_DEFAULT_TEMPLATED_HTML_CONTENT_PATH` is refers to the provider data

SMTP_DEFAULT_TEMPLATED_SUBJECT also could be defined there

Reply:
I guess this constants added into the core settings by a mistake in https://github.com/apache/airflow/pull/36226
IMHO, instead of that we should use data from the smtp provider and avoid core settings.

And even more, we should deprecate this into the `settings.py` and suggest to upgrade to the never provider

Reply:
My point here, that we do not have to use `airflow.settings` for provide some functional for providers, except the cases when we move something from core to providers, e.g. executors.

If we want to make it configurable, then we need to utilise Airflow Configurations instead

Reply:
> If we want to make it configurable, then we need to utilise Airflow Configurations instead

Yep. Absolutely we should do it this way. @vchiapaikeo -> could you please follow up on that one and open a new PR moving the change to smtp provider configuration (with deprecation) ?

Reply:
I could create a task and someone could pick it up.

Reply:
https://github.com/apache/airflow/issues/37703